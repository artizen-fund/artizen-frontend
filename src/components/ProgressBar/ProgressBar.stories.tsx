import styled from 'styled-components'
import ProgressBar, { ProgressBarProps } from './'

export default {
  title: 'components/ProgressBar',
  component: ProgressBar,
  argTypes: {
    children: {
      control: 'number',
    },
    noSmoothing: {
      control: 'boolean',
    },
  },
}

// note: this is just here so the progressbar doesn't go full-screen-width
const Wrapper = styled.div`
  position: relative;
  max-width: 500px;
`

export const ProgressBarComponent = (props: ProgressBarProps) => {
  return (
    <Wrapper>
      <ProgressBar {...props} />
    </Wrapper>
  )
}
