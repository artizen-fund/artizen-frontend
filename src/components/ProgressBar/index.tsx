import styled from 'styled-components'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

export interface ProgressBarProps {
  children: number
  noSmoothing?: boolean
}

const ProgressBar = styled.div<ProgressBarProps>`
  height: 4px;
  background-color: ${rgba(palette.stone, 0.24)};
  @media (prefers-color-scheme: dark) {
    background-color: ${rgba(palette.barracuda, 0.24)};
  }
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    height: 6px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    height: 8px;
  }
  border-radius: 9999px;
  color: transparent;

  &:after {
    position: absolute;
    display: block;
    left: 0;
    top: 0;
    height: 100%;
    width: ${props => props.children}%;
    max-width: 100%;

    content: '${props => props.children}%';
    border-radius: 9999px;
    background-color: ${rgba(palette.algae)};
    ${props =>
      !props.noSmoothing &&
      `
      transition: width 0.3s ease-in-out;
    `}
  }
`

export default ProgressBar
