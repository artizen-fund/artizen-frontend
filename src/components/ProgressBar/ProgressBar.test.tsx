import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import ProgressBar from './'

describe('ProgressBar', () => {
  it('renders ProgressBar unchanged', () => {
    const { container } = render(<ProgressBar>{50}</ProgressBar>)
    expect(container).toMatchSnapshot()
  })
})
