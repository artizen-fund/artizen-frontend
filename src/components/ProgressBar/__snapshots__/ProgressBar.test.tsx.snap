// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders ProgressBar unchanged 1`] = `
<div>
  <div
    class="sc-d725370f-0 gBakkx"
  >
    50
  </div>
</div>
`;
